src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -659,6 +664,27 @@ private void MakeExplicitParameterTypeInferences(Binder binder, BoundExpression
                 MakeExplicitParameterTypeInferences(binder, element, targetElementType, kind, ref useSiteInfo);
             }
         }
+
+        private void MakeSpreadElementTypeInferences(
+            BoundCollectionExpressionSpreadElement argument,
+            TypeWithAnnotations target,
+            ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            Debug.Assert(target.Type is { });
+            if (target.Type is null)
+            {
+                return;
+            }
+
+            var enumeratorInfo = argument.EnumeratorInfoOpt;
+            if (enumeratorInfo is null)
+            {
+                return;
+            }
+
+            // https://github.com/dotnet/roslyn/issues/68786: Ignoring top-level nullability of enumeratorInfo.ElementType.
+            LowerBoundInference(TypeWithAnnotations.Create(enumeratorInfo.ElementType), target, ref useSiteInfo);
```

Comment:
Why not use `ElementTypeWithAnnotations`?

Reply:
Seems like this could break something and cause us to not infer a type as nullable where we should. Please make sure to test something like:
```cs
M([..[default(string?)]]);
void M<T>(T[] arr);
```

Reply:
Thanks, I hadn't noticed `ElementTypeWithAnnotations` previously.

I've added a couple of tests for inferring nullability but we are still not handling collection expressions in NullableWalker. If possible, I'd prefer to handle this when fixing [https://github.com/dotnet/roslyn/issues/68786](https://github.com/dotnet/roslyn/issues/68786).